dietpi/dietpi-software

```diff
@@ -12318,6 +12324,40 @@ _EOF_
 			Download_Install "$(curl -sSfL 'https://api.github.com/repos/rclone/rclone/releases/latest' | mawk -F\" "/\"browser_download_url\": .*\/rclone-v[^\"\/]*linux-$arch.deb\"/{print \$4}")"
 
 		fi
+		
+		software_id=206 # Homer
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+		
+		Banner_Installing
+		
+		# installing nginx and ensuring it starts at boot
+		sudo apt install nginx -y
+		sudo systemctl enable nginx
+		
+		# installing homer
+		wget "https://github.com/bastienwirtz/homer/releases/latest/download/homer.zip" 
+		sudo unzip homer.zip -d /mnt/dietpi_userdata/Homer/
+		sudo cp /usr/share/nginx/homer/assets/config.yml.dist /mnt/dietpi_userdata/Homer/assets/config.yml
```

Comment:
We have in install function that will download and unzip automatically. Have a look into the script for `Download_Install`. As well `sudo` is not needed as our software script will be running as `root` user. Next to this we use an own error handle to manage issues during install. Check for `G_EXEC` function.

Reply:
It could look like this:
```sh
Download_Install 'https://github.com/bastienwirtz/homer/releases/latest/download/homer.zip' /mnt/dietpi_userdata/homer
```

Are you sure that `/usr/share/nginx/homer/assets/config.yml.dist` exists? I cannot see it in the `nginx` or `nginx-common` package contents 🤔.

Reply:
> One thing that is missing is how to handle reinstalls. In this case you don't want the existing config to be overwritten. As static pages. Homer cannot update itself, right? If I see it correctly, it is intended that custom files (like icons and such) are added, so it is not feasible to copy each custom file (if existent) into a newly downloaded Homer instance, but instead it seems to be safe to extract the newly downloaded archive into existing Homer directories, as it contains only samples, right?

yeah the issue is we don't know what the user named the directories/files for stuff like icons/wallpapers etc, should i make the installer interactive and have the user enter named to allow updates?

Reply:
maybe we should simply go to overwrite existing homer folder by extracting zip directly and don't try to migrate user data.

Reply:
> `G_EXEC`

can this be used on any command?

Reply:
yes, we use it on every step. Just check out other install options.

Reply:
> maybe we should simply go to overwrite existing homer folder by extracting zip directly and don't try to migrate user data.

i'll make it copy data if it already exists to /tmp, to allow recovery, and then overwrite files